presto-common/src/main/java/com/facebook/presto/common/type/CharType.java

```diff
@@ -48,7 +47,7 @@ private CharType(long length)
                 Slice.class);
 
         if (length < 0 || length > MAX_LENGTH) {
-            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("CHAR length scale must be in range [0, %s]", MAX_LENGTH));
+            throw new InvalidFunctionArgumentException(format("CHAR length scale must be in range [0, %s]", MAX_LENGTH));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: MAX_LENGTH. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 50. Consider formatting this data with &#39;%d&#39; instead.